core/domain/platform_parameter_registry_test.py

```diff
@@ -234,6 +234,87 @@ def test_update_parameter_with_invalid_rules_failure(self):
                 ],
             )
 
+    def test_update_dev_feature_with_rule_enabled_for_test_raises_exception(
+            self):
+        parameter_name = 'parameter_a'
+        registry.Registry.create_feature_flag(
+            parameter_name, 'dev feature', FEATURE_STAGES.dev)
+
+        with self.assertRaisesRegexp(
+            utils.ValidationError,
+            'Feature in dev stage cannot be enabled in test or production '
+            'environment.'):
+            registry.Registry.update_platform_parameter(
+                parameter_name,
+                feconf.SYSTEM_COMMITTER_ID,
+                'commit message',
+                [
+                    {
+                        'filters': [
+                            {
+                                'type': 'server_mode',
+                                'conditions': [('=', FEATURE_STAGES.test)]
+                            }
+                        ],
+                        'value_when_matched': True
+                    }
+                ],
+            )
+
+    def test_update_dev_feature_with_rule_enabled_for_prod_raises_exception(
+            self):
+        parameter_name = 'parameter_a'
+        registry.Registry.create_feature_flag(
+            parameter_name, 'dev feature', FEATURE_STAGES.dev)
+
+        with self.assertRaisesRegexp(
+            utils.ValidationError,
+            'Feature in dev stage cannot be enabled in test or production '
+            'environment.'):
```

Comment:
This validation error should probably say ``in test or production environments.`` (typo nit for 'environment')

Reply:
Done.